packages/react-components/components/PhoneNumberInput.tsx

```diff
@@ -65,6 +69,37 @@ export default class PhoneNumberInput extends React.Component<Props, State> {
     }
   }
 
+  async triggerPhoneNumberRequest() {
+    try {
+      const phone = await SmsRetriever.requestPhoneNumber()
+      const phoneNumber = parsePhoneNumber(phone, '')
+
+      if (phoneNumber) {
+        this.setState({
+          phoneNumber: phoneNumber.displayNumber.toString(),
+        })
+        if (phoneNumber.countryCode) {
+          // TODO known issue, the country code is not enough to
+          // get a country, e.g. +1 could be USA or Canada
+          const displayName = this.state.countries.getCountryByPhoneCountryCode(
+            '+' + phoneNumber.countryCode.toString()
+          ).displayName
+
+          this.onChangeCountryQuery(displayName)
+        }
+      }
+    } catch (error) {
+      console.debug(`${TAG}/Could not request phone`)
+    }
+  }
+
+  onCountryFocus = async () => {
+    if (this.props.onInputFocus) {
+      await this.props.onInputFocus()
+    }
+    await this.triggerPhoneNumberRequest()
```

Comment:
If this functionality only works on android, you should probably check the platform here and only call this if we're on android